ocaml/tests/test_helpers.ml

```diff
@@ -347,6 +347,50 @@ module CIDRCheckers = Generic.MakeStateless (struct
       ]
 end)
 
+module RunInParallel = Generic.MakeStateless (struct
+  module Io = struct
+    type input_t = (unit -> string) list * int
+
+    type output_t = (string list, exn) result
+
+    let string_of_input_t (_, c) = string_of_int c
+
+    let string_of_output_t = Fmt.(str "%a" Dump.(result ~ok:(list string) ~error:exn))
+  end
+
+  let transform (funs, capacity) =
+    try Ok (Helpers.run_in_parallel ~funs ~capacity)
+    with e -> Error e
+
+  let f () = "hello"
+
+  exception Unittest_exception
+
+  let fe () =
+    match f () with
+    | "not-hello" -> "hello"
+    | _ -> raise Unittest_exception
+
+  let tests =
+    `QuickAndAutoDocumented
+      [
+          ( ([], 0), Ok [] )
+        ; ( ([f], 0), Ok [] )
+        ; ( ([f], 1), Ok ["hello"] )
+        ; ( ([f], 2), Ok ["hello"] )
+        ; ( ([f; f], 1), Ok ["hello"; "hello"] )
+        ; ( ([f; f], 2), Ok ["hello"; "hello"] )
+        ; ( ([f; f], 3), Ok ["hello"; "hello"] )
+        ; ( ((List.init 64 (fun _ -> f)), 64), Ok (List.init 64 (fun _ -> "hello")) )
```

Comment:
A stronger test would be something like this (to ensure that `run_in_parallel` preserves the order of its arguments)

```suggestion
        ; ( let range_64 = List.init 64 Fun.id |> List.map string_of_int in
            ( (range_64, 64), Ok range_64 ) )
```

Reply:
`let range_64 = List.init 64 string_of_int in` seems clearer to me ;)